tests/models/test_modeling_common.py

```diff
@@ -862,6 +865,29 @@ def test_model_parallelism(self):
 
                 self.assertTrue(torch.allclose(base_output[0], new_output[0], atol=1e-5))
 
+    @require_torch_gpu
+    def test_sharded_checkpoints(self):
```

Comment:
It could be helpful to have a similar test but for models stored on the Hub. Not sure if the model could be pushed to the Hub without exposing the token, but at least the test could load a sharded model from the Hub.

Reply:
Done in d144526f0.